var/spack/repos/builtin/packages/hip/package.py

```diff
@@ -66,7 +67,7 @@ def patch(self):
             'hip-config.cmake.in', string=True)
 
     def flag_handler(self, name, flags):
-        if name == 'cxxflags' and '@3.7.0' in self.spec:
+        if name == 'cxxflags' and not ('@3.5.0' in self.spec):
```

Comment:
You could also use:
```suggestion
        if name == 'cxxflags' and '@3.7.0:' in self.spec:
```
The trailing colon means an open-ended version range.